### PR TITLE
fix(DB): Portal: Karazhan usable in battlegrounds

### PR DESCRIPTION
**Pull request by Hellcat404**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/18173

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [x] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by Nyeriah** [comment]
Might not be possible to check but there's a spell attribute to prevent casting in pvp zones. Depending on the message sent to client when spell fails maybe that should be used instead of disable. I dont remember how disables handle this specifically tbh

**Comment by Hellcat404** [comment]
After taking a look, I think you're right - I'll make a change in conditions to give a spell cast result instead of using disables.

I saw the ritual of summoning disable in the database but I think that should also be moved to conditions

**Review by elthehablo — CHANGES_REQUESTED** [review]
I don't understand why the update query is there? better to just delete and insert the whole line

File: data/sql/updates/pending_db_world/rev_1705323899511928342.sql [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES (17, 0, 28148, 0, 0, 22, 0, 607, 0, 0, 1, 166, 0, '', 'Portal: Karazhan - Restrict Strand of the Ancients');
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=17 AND `SourceGroup`=0 AND `SourceEntry`=28148 AND `SourceId`=0 AND `ElseGroup`=0 AND `ConditionTypeOrReference`=22 AND `ConditionTarget`=0 AND `ConditionValue1`=628 AND `ConditionValue2`=0 AND `ConditionValue3`=0;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES (17, 0, 28148, 0, 0, 22, 0, 628, 0, 0, 1, 166, 0, '', 'Portal: Karazhan - Restrict Isle of Conquest');
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1705323899511928342.sql:25]: https://www.azerothcore.org/wiki/sql-standards#compact-queries

File: data/sql/updates/pending_db_world/rev_1705323899511928342.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES (17, 0, 698, 0, 0, 22, 0, 30, 0, 0, 1, 166, 0, '', 'Ritual of Summoning - Restrict Alterac Valley');
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=17 AND `SourceGroup`=0 AND `SourceEntry`=698 AND `SourceId`=0 AND `ElseGroup`=0 AND `ConditionTypeOrReference`=22 AND `ConditionTarget`=0 AND `ConditionValue1`=489 AND `ConditionValue2`=0 AND `ConditionValue3`=0;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES (17, 0, 698, 0, 0, 22, 0, 489, 0, 0, 1, 166, 0, '', 'Ritual of Summoning - Restrict Warsong Gluch');
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1705323899511928342.sql:5]: Typo

File: data/sql/updates/pending_db_world/rev_1705323899511928342.sql [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES (17, 0, 28148, 0, 0, 22, 0, 30, 0, 0, 1, 166, 0, '', 'Portal: Karazhan - Restrict Alterac Valley');
+DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId`=17 AND `SourceGroup`=0 AND `SourceEntry`=28148 AND `SourceId`=0 AND `ElseGroup`=0 AND `ConditionTypeOrReference`=22 AND `ConditionTarget`=0 AND `ConditionValue1`=489 AND `ConditionValue2`=0 AND `ConditionValue3`=0;
+INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES (17, 0, 28148, 0, 0, 22, 0, 489, 0, 0, 1, 166, 0, '', 'Portal: Karazhan - Restrict Warsong Gluch');
```
Comment (elthehablo) [inline data/sql/updates/pending_db_world/rev_1705323899511928342.sql:17]: Another typo

**Review by elthehablo — CHANGES_REQUESTED** [review]
You can put the delete statements together with an in on the map ids to clean up the file and do an insert with multiple lines

```sql
INSERT INTO TABLE VALUES
(something),
(something),
(something);
```

**Comment by pangolp** [comment]
Update the SQL file. But I'm not sure how to test this.

**Pull request closed without merging**